src/main/js/components/SignUp.js

```diff
@@ -0,0 +1,76 @@
+import React, {useContext} from 'react';
+import {firebaseAuth} from "../services/provider/AuthProvider";
+import {withRouter} from 'react-router-dom';
+import {Button, FormControl, Grid, Input, InputLabel} from "@material-ui/core";
+import {makeStyles} from "@material-ui/core/styles";
+
+const useStyles = makeStyles((theme) => ({
+    root: {
+        display: 'flex',
+        flexWrap: 'wrap',
+    },
+    margin: {
+        margin: theme.spacing(1),
+    },
+    headLine: {
+        textAlign: "center"
+    },
+    centerBlock: {
+        margin: '0 auto'
+    }
+}));
+
+const SignUp = (props) => {
+    const classes = useStyles();
+    const {handleSignUp, inputs, setInputs, errors} = useContext(firebaseAuth)
+
+    const handleSubmit = async (e) => {
+        e.preventDefault()
+        await handleSignUp()
+        props.history.push('/')
+    }
+    const handleChange = e => {
+        const {name, value} = e.target
+        setInputs(prev => ({...prev, [name]: value}))
+    }
+
+    return (
+        <div className={classes.root}>
+            <form onSubmit={handleSubmit} className={classes.centerBlock}>
+                <Grid container spacing={3}>
+                    <Grid item xs className={classes.paper}>
+                        <h1 className={classes.headLine} >
+                            Sign Up
+                        </h1>
+                    </Grid>
+                </Grid>
+                <Grid container spacing={3}>
+                    <Grid item xs>
+                        <FormControl fullWidth className={classes.margin}>
+                            <InputLabel htmlFor="user-login">Email</InputLabel>
+                            <Input
+                                id="user-login"
+                                value={inputs.email}
+                                onChange={handleChange}
+                                name="email"
+                            />
+                        </FormControl>
+                        <FormControl fullWidth className={classes.margin}>
+                            <InputLabel htmlFor="user-pass">Password</InputLabel>
+                            <Input
+                                id="user-pass"
+                                value={inputs.password}
+                                onChange={handleChange}
+                                name="password"
```

Comment:
Echoing previous comments so we don't forget: adding type="password" here to make the text hidden on entry to the user

Reply:
Will change that. Thank you.